typescript/cli/src/commands/warp.ts

```diff
@@ -366,3 +287,44 @@ const send: CommandModuleWithWriteContext<
     process.exit(0);
   },
 };
+
+export const check: CommandModuleWithContext<{
+  config: string;
+  symbol?: string;
+  warp?: string;
+}> = {
+  command: 'check',
+  describe:
+    'Verifies that a warp route configuration matches the on chain configuration.',
+  builder: {
+    symbol: {
+      ...symbolCommandOption,
+      demandOption: false,
+    },
+    warp: {
+      ...warpCoreConfigCommandOption,
+      demandOption: false,
+    },
+    config: inputFileCommandOption({
+      defaultPath: DEFAULT_WARP_ROUTE_DEPLOYMENT_CONFIG_PATH,
+      description: 'The path to a warp route deployment configuration file',
+    }),
+  },
+  handler: async ({ context, config, symbol, warp }) => {
+    logCommandHeader('Hyperlane Warp Check');
+
+    const warpRouteConfig = await readWarpRouteDeployConfig(config, context);
+    const onChainWarpConfig = await runWarpRouteRead({
+      context,
+      warp,
+      symbol,
+    });
+
+    await runWarpRouteCheck({
```

Comment:
nit: can be useful to output to a diff file for those with big configs and little scroll back